File: extensions/panache/hibernate-panache/deployment/src/test/java/io/quarkus/hibernate/panache/deployment/test/MyEntity.java
package io.quarkus.hibernate.panache.deployment.test;

import java.util.List;

import jakarta.persistence.Entity;

import org.hibernate.annotations.processing.Find;
import org.hibernate.annotations.processing.HQL;

import io.quarkus.hibernate.panache.PanacheEntity;
import io.quarkus.hibernate.panache.PanacheRepository;

@Entity
public class MyEntity extends PanacheEntity {
    public String foo;
    public String bar;

    interface ManagedBlockingQueries extends PanacheRepository<MyEntity> {
        default List<MyEntity> findFoos(String val) {
            return list("foo", val);
        }

        @HQL("where foo = :val")
        List<MyEntity> findFoosHQL(String val);

        @Find
        List<MyEntity> findFoosFind(String foo);
    }
}

File: extensions/panache/hibernate-reactive-panache-kotlin/deployment/src/test/kotlin/io/quarkus/hibernate/reactive/panache/kotlin/deployment/test/MyEntity.kt
package io.quarkus.hibernate.reactive.panache.kotlin.deployment.test

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheEntity
import jakarta.persistence.Entity

@Entity
class MyEntity : PanacheEntity() {
    companion object: PanacheCompanion<MyEntity> {
    }

    lateinit var name: String
}
